File: clouditor-engine-k8s/src/main/java/io/clouditor/discovery/k8s/KubernetesPodScanner.java
package io.clouditor.discovery.k8s;

import io.clouditor.discovery.ScanException;
import io.clouditor.discovery.ScannerInfo;
import io.kubernetes.client.ApiException;
import io.kubernetes.client.models.V1Pod;
import java.util.List;

@ScannerInfo(assetType = "Pod", group = "Kubernetes", service = "Compute")
public class KubernetesPodScanner extends KubernetesScanner<V1Pod> {

  public KubernetesPodScanner() {
    super(x -> x.getMetadata().getUid(), x -> x.getMetadata().getName());
  }

  @Override
  protected List<V1Pod> list() throws ScanException {
    try {
      var list = this.api.listPodForAllNamespaces(null, null, null, null, null, null, null, null);

      return list.getItems();
    } catch (ApiException e) {
      throw new ScanException(e);
    }
  }
}

File: clouditor-engine-k8s/src/main/java/io/clouditor/discovery/k8s/KubernetesPodScanner.kt
package io.clouditor.discovery.k8s

import io.clouditor.discovery.ScanException
import io.clouditor.discovery.ScannerInfo
import io.kubernetes.client.ApiException
import io.kubernetes.client.models.V1Pod
import java.util.function.Function

@ScannerInfo(assetType = "Pod", group = "Kubernetes", service = "Compute")
class KubernetesPodScanner :
    KubernetesScanner<V1Pod>(Function { it.metadata.uid }, Function { it.metadata.name }) {
    @Throws(ScanException::class)

    override fun list(): List<V1Pod> {
        return try {
            val list = api!!.listPodForAllNamespaces(null, null, null, null, null, null, null, null)
            list.items
        } catch (e: ApiException) {
            throw ScanException(e)
        }
    }
}